server/src/main/java/io/pravega/schemaregistry/service/SchemaRegistryService.java

```diff
@@ -896,11 +896,18 @@ private SchemaInfo normalizeSchemaBinary(SchemaInfo schemaInfo) {
                     // treated to be equal. 
                     JsonNode jsonNode = OBJECT_MAPPER.readTree(schemaString);
                     Object obj = OBJECT_MAPPER.treeToValue(jsonNode, Object.class);
+
```

Comment:
Initialize type to NameUtil.qualifiedName(namespace, group) if `schemaInfo.getType();` @ line 820 is null/empty.

Reply:
a new method created to create type if the same is not provided with the API call. Implementation of method is applied with JSON and Custom, for Proto and Avro we already have the implementation

Reply:
a new method created to create type if the same is not provided with the API call. Implementation of method is applied with JSON and Custom, for Proto and Avro we already have the implementation